scan/lib/scan/xcpretty_reporter_options_generator.rb

```diff
@@ -55,7 +55,8 @@ def generate_reporter_options
       # adds another junit reporter in case the user does not specify one
       # this will be used to generate a results table and then discarded
       require 'tempfile'
-      Scan.cache[:temp_junit_report] = Tempfile.new("junit_report").path
+      @temp_junit_report = Tempfile.new("junit_report")
```

Comment:
I'm not really sure what you mean by pipe here?

Here was my thought for why we're okay regarding garbage collection.
```
Scan::Runner has an instance of TestCommandGenerator
TestCommandGenerator has an instance of XCPrettyReporterOptionsGenerator
XCPrettyReporterOptionsGenerator has an instance of Tempfile
```